library/src/main/java/org/apache/apex/malhar/lib/state/managed/MovingBoundaryTimeBucketAssigner.java

```diff
@@ -125,9 +125,11 @@ public long getTimeBucket(long time)
       long move = (diffInBuckets + 1) * bucketSpanMillis;
       start += move;
       end += move;
-      lowestPurgeableTimeBucket += diffInBuckets;
       // trigger purge when lower bound changes
-      triggerPurge = (diffInBuckets > 0);
+      triggerPurge = (diffInBuckets >= 0);
```

Comment:
This is actually because in this case, the window is start to end - 1
Its little un-intuitive, since at the beginning, we have initialized `end = start + (numBuckets * bucketSpanMillis)` instead of `end = start + (numBuckets * bucketSpanMillis) - 1`
@chaithu14 Please add to the documentation of this method.